Phase1 20210907

eoliveros opened the pull request:
- [x] added functions in reporting_endpoint for the dashboard broker order
- [x]  added dashboard for broker order. 
- [x] the dashboard_broker_order is now expanding the row when new market is added.
- [x] no logout button when login, it just have login. so you cannot logout.
- [x] dashboard link when login is going to the index. so its not going anywhere.
- [x] ability to choose any of the avaliable themes from bootswatch. this is now done via host env. So, if you don't have anything in the host_env for the "FLASK_ADMIN_SWATCH" it goes to the "default" theme. The default theme is the light version. The "slate" theme is the dark version.
- [x] loading a different version(light/dark) of css depending on the theme.
- [x] dashboard_general still showing and using information from premio/paydb. premio needs to be removed. meaning need to move it away from it.
- [x] each asset has a different unit. 1 bitcoin = 100000000 satoshi, 1 eth = 1000000000000000000 wei, etc... need to create something in the reporting that would do this automatically rather than statically assigning it in the function. so that when new assets are added, no changes in reporting required.
- [x] the columns on the dashboard flexes(increases/decreases in size) and is independent of the row. therefore if one column gets too big, its no longer aligning with the rest it will make the rest of the column in the row smaller and no longer aligns to the columns. this will require some changing to what columns to show. changed the layout of each boxes. its now has tx count on the top and the bottom is the amount of the asset. and its smaller.
oct reporting:
- [x] change the frequency variable to lambda functions. to fix the issue with the time drift as time goes on, the further the value of the boxes becomes.
- [x] fix the issue where the amount on dashboard is replaced by zero's instead of the value.
- [x] add date filter for broker_order
- [x] add asset filter for broker order
- [x] create a link in the dashboard_broker_order and map it to the admin page with the applied filters.

Cleanups:
- [x] Remove the dashboard_proposals.html from the rest of the dashboard pages
- [x] Remove the dashboard_paydb_premio.html from the rest of the dashboard pages
- [x] Remove the url link from all of the remaining dashboard pages for the proposals page.
- [x] Remove the unused code in reporting_endpoint.py

djpnewton commented:
can you fill out the description of the PR describing the changes that have been made

djpnewton commented:
>     * overwrite the flask-security email for reset password.

Do you mean 'update the reset password template'?

> 
>     * apply the theme

what theme?

> 
>     * changes to the index page

what changes to the index page?

> 
>     * changes to the first page after login.

what changes?

> 
>     * added dashboard for broker order(incomplete). need to know what assets.

ok 

> 
>     * added functions in reporting_endpoint for the dashboard broker order

ok

> 
>     * will have to be re-aligned for later (dashboard)

what???

djpnewton reviewed (changes requested):
I think you should just focus on the broker orders dashboard and leave out the other stuff from this PR. That change will be big enough to develop and review

eoliveros commented on reporting_endpoint.py:
```diff
@@ -33,6 +33,46 @@
 LAST_DAY_CURRENT_YEAR = FIRST_DAY_NEXT_YEAR - timedelta(days=1)
 
+def report_dashboard_broker_order():
+    btc_order_count_today = broker_order_count(BrokerOrder, TODAY, TOMORROW, 'BTC-NZD')
```
i changed this to be like what you have.

eoliveros replied:
already changed this on my push, not sure what happened to my previous comment about me making the change.

eoliveros commented:
> > ```
> > * overwrite the flask-security email for reset password.
> > ```
> 
> Do you mean 'update the reset password template'?
> 
yes
> > ```
> > * apply the theme
> > ```
> 
> what theme?
> 
using the slate theme. also i made it so that the theme can be changed by using environments... at this very moment however, looks like the log_user.html has been overwritten to the one in flask-security. The dark theme does not apply since everything is black. it pretty much looks like its blank/black page with a navbar on the top.
> > ```
> > * changes to the index page
> > ```
> 
> what changes to the index page?
> 
i deleted somethings in there ie. push notification since thats not needed. Also was applying themes on the buttons so that depending on the light/dark theme, it will try to match it. no longer the case, its just now white.
The url_for for the dashboard was broken. just fixed it. 
> > ```
> > * changes to the first page after login.
> > ```
> 
> what changes?
> 
can't remember now, but this was overwritten with the pull i just did from the github
> > ```
> > * added dashboard for broker order(incomplete). need to know what assets.
> > ```
> 
> ok
> 
This should now add more rows corresponding the new market.
> > ```
> > * added functions in reporting_endpoint for the dashboard broker order
> > ```
> 
> ok
> 
So created some functions within reporting_endpoint for the dashboard_broker_order
> > ```
> > * will have to be re-aligned for later (dashboard)
> > ```
> 
> what???
because the each row has an independent column, and the columns flexes, the more data you have on the column, the bigger the column gets. which means the column above/below the row that's updated will no longer match the column (above/below the row). look image below:
![image](https://user-images.githubusercontent.com/47799652/132448955-4fe22476-aad9-42f9-aaaf-90f65dd4d31c.png)
this is what ends up happening.

djpnewton commented:
> i deleted somethings in there ie. push notification since thats not needed. Also was applying themes on the buttons so that depending on the light/dark theme, it will try to match it. no longer the case, its just now white.
> The url_for for the dashboard was broken. just fixed it.

Leave the push notification there for now, we could use it later.

Themes should be applied by changing the CSS file used not switch statements on the element classes

djpnewton commented on templates/index.html:
```diff
@@ -2,10 +2,11 @@
 {% block content %}
 {% if current_user.is_authenticated %}
-    <div class='some-page-wrapper'>
+    <div class='{{ config["DASHBOARD_PAGE_WRAPPER"] }}'>
```
dont do this, change the css file used instead

eoliveros replied:
i could create two css and change the names of the field to be the same from boht files but different colors.

eoliveros replied:
two different css based on might/dark themes

djpnewton commented on templates/index.html:
```diff
@@ -2,10 +2,11 @@
         <div class='row'>
             {% if current_user.has_role("admin") or current_user.has_role("finance") %}
-            <a href="{{ url_for('index') }}">
+	    <!--<a href="{{ url_for('index') }}">-->
```
dead code

eoliveros replied:
i removed this yesterday but maybe after the code was checked.

eoliveros replied:
re-pushed without this in the index,.html

djpnewton commented on templates/layout.html:
```diff
@@ -12,17 +12,52 @@
     <link rel="stylesheet" href="{{ url_for('static', filename='css/custom_reporting.css') }}"crossorigin="anonymous"/>
 
     <title>{% block title %}{% endblock %}</title>
+    {% if config["FLASK_ADMIN_SWATCH"] == "slate" -%}
```
dont do if/else statements like this on the html elements just for themes

something like will be much easier to manage:
```html
<link rel="stylesheet" href="{{ url_for('static', filename='css/custom_reporting.css') }}"crossorigin="anonymous"/>
{% if config['CSS_THEME'] %}
    <link rel="stylesheet" href="{{ url_for('static', filename=config['CSS_THEME']) }}"crossorigin="anonymous"/>
{% endif %}
```

eoliveros replied:
Hmmm.... the custom_reporting.css file has got nothing to do with this particular page. The nav class used here is what is in the examples from the bootswatch, not the custom_reporting.css.

eoliveros replied:
ive tried adding the navbar to the custom_reporting.css files but i cant get it to work at all. so i cant figure a way of getting removing if condition for the.

eoliveros replied:
im applying the config['CSS_THEME'] but cant figure a way to do this with the navbar part.

djpnewton replied:
you dont need to change the bootstrap templates you can control them with the FLASK_ADMIN_SWATCH param (https://github.com/flask-admin/flask-admin/blob/e39f786374ce0e60db93f583efacb4672de0025c/flask_admin/templates/bootstrap3/admin/base.html#L15)

As for the changing the navbar bootstrap styles if you really want to change them you could do something like:

```html
...
<nav class="navbar navbar-expand-lg navbar-{{ config['CSS_THEME_INTENSITY'] }} bg-{{ config['CSS_THEME_INTENSITY'] }}">
...
```

eoliveros replied:
changed this so its using the intensity.

eoliveros commented:
merge from upstream master to my branch and re-added everything i could see thats not there.

djpnewton commented:
is this ready for me to look at?

eoliveros commented:
> is this ready for me to look at?

yes, it should be.

djpnewton commented:
I am getting an error loading the template:
```
  File "/Users/daniel/projects/beryllium/templates/reporting/dashboard_general.html", line 14, in block "content"
    {% for balance in dasset_balances %}
TypeError: 'NoneType' object is not iterable
```

djpnewton commented on reporting_endpoint.py:
```diff
@@ -109,107 +76,63 @@ def report_user_balance():
-        result = 0
-    return result
+def get_decimals(asset):
+    dec = eval('dasset.'+str(asset)+'.decimals')+1
```
why are you using eval here?

eoliveros replied:
> why are you using eval here?

This is the only way I could substitute the word "asset" with whatever asset and have the dasset.py return me the decimals.

eoliveros replied:
since im able to use the other function, this is no longer needed.

eoliveros commented:
> I am getting an error loading the template:
> 
> ```
>   File "/Users/daniel/projects/beryllium/templates/reporting/dashboard_general.html", line 14, in block "content"
>     {% for balance in dasset_balances %}
> TypeError: 'NoneType' object is not iterable
> ```

not sure why you're having that issue... i deployed my branch to premio-erick and its loading.

djpnewton commented on reporting_endpoint.py:
```diff
@@ -109,107 +76,63 @@ def report_user_balance():
-    if not result:
-        result = 0
-    return result
+def get_decimals(asset):
```
you can get rid of this function and just use `dasset.asset_int_to_dec` instead

eoliveros replied:
used the above

djpnewton commented on reporting_endpoint.py:
```diff
@@ -109,107 +76,63 @@ def report_user_balance():
-    if not result:
-        result = 0
-    result = utils.int2asset(result)
+def asset_divisor(data, divisor):
```
get rid of this function too

eoliveros replied:
deleted the function

djpnewton commented on templates/reporting/dashboard_broker_orders.html:
```diff
@@ -0,0 +1,114 @@
+    <div class='column'>
+      <a href="#">
+      <div class='report-row report-row-hover white text_color'>
+        {{ orders_data[market].order_count_today }}({{ orders_data[market].order_amount_today }} {{ orders_data[market].asset_symbol }})
```
```suggestion
        {{ orders_data[market].order_count_today }}<br/>{{ orders_data[market].order_amount_today }} {{ orders_data[market].asset_symbol }}
```

also for all the rest

eoliveros replied:
included a break to make it 2 lines instead of one.

djpnewton commented on static/css/custom_reporting_dark.css:
```diff
@@ -0,0 +1,209 @@
+  height: 100px;
+  margin: 5px;
+  text-align: center;
+  padding: 30px;
```
```suggestion

```

eoliveros replied:
i cant find a way to vertically align the what is in the column for flex css.
padding seems to be the only way to somehow make it look like its in the middle.

eoliveros replied:
changed the way its applied instead of delete

djpnewton commented on static/css/custom_reporting_dark.css:
```diff
@@ -0,0 +1,209 @@
+  margin: 5px;
+  text-align: center;
+  padding: 30px;
+  font-size: 150%;
```
```suggestion

```

eoliveros replied:
deleted this line.

djpnewton reviewed (changes requested):
basically you should be able to make the text look better in the broker order boxes

djpnewton commented:
i have made some changes.

Take a look at how I changed `custom_reporting_dark.css`. I dont want to see you copy a 200 line css file just to change a couple of colors. You keep the original css file and then add the new one after which *only overrides the new colors etc for that theme*

djpnewton commented:
this PR is not passing the lint checks

eoliveros commented:
not sure how to fix this pylint error. appears to be the last one:
```
************* Module dasset
dasset.py:483:8: W0622: Redefining built-in 'id' (redefined-builtin)
```

djpnewton commented:
> not sure how to fix this pylint error. appears to be the last one:
> 
> ```
> ************* Module dasset
> dasset.py:483:8: W0622: Redefining built-in 'id' (redefined-builtin)
> ```

dont worry about that one, that was my fault

eoliveros commented on templates/index.html:
```diff
@@ -1,35 +1,37 @@
+{{ super() }}
 
 {% block content %}
+{{ super() }}
```
deleted. was trying to figure something out as to why the body is not applying the color

eoliveros commented on templates/index.html:
```diff
@@ -1,35 +1,37 @@
 {% extends "layout.html" %}
+{{ super() }}
```
deleted. was trying to figure something out as to why the body is not applying the color

djpnewton commented on templates/security/email/reset_instructions.html:
```diff
@@ -12,4 +12,4 @@
 
-{% endblock %}
\ No newline at end of file
+{% endblock %}
```
this template should not be changed - it has nothing to do with reporting

eoliveros replied:
looks like my merging may have deleted this. or before the merge it was deleted from the master?

eoliveros replied:
ahhh no. i had my own version of reset_instruction.html and in the master it was rewritten to be shorter than my html thats why it was changed.

djpnewton replied:
remove the irrelevant change from your PR

eoliveros replied:
the entire file was overwritten, its not a line change.

djpnewton replied:
then make another commit to revert that one line change

djpnewton replied:
fine I will do it myself: https://github.com/zap-me/beryllium/pull/1/commits/f6a81d0f895d2110878840cc64011e2faadbec21

eoliveros commented:
vi the reset_instructions.html file:
```
    <p><a href="{{ reset_link }}">{{ _fsdomain('Click here to reset your password') }}</a></p>

{% endblock %}
~
~
~
```

So there's no line after the endblock. Can't delete anything in there. I'll delete the endblock and readd it again and see if it works.

djpnewton commented:
looks like the amounts are not correct (all 0) and the links are not clickable:
![image](https://user-images.githubusercontent.com/42459/134288212-7a574646-d79e-4241-8b8f-169b7e35e29c.png)

eoliveros commented:
the links are not clickable because there's no where to link it to. as for the 0's not sure, need to have a look.

djpnewton commented:
> the links are not clickable because there's no where to link it to. as for the 0's not sure, need to have a look.

it could link to the brokerorder table in the admin view with filters around date and base_asset

eoliveros commented:
> > the links are not clickable because there's no where to link it to. as for the 0's not sure, need to have a look.
> 
> it could link to the brokerorder table in the admin view with filters around date and base_asset

will need to create the filters first to do that. i'll look into it, after i figure out why the amount is returning 0's instead of the values.

eoliveros commented:
just pulled the update. everytime i refresh the dashboard on broker order, im getting the following:
```
sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: broker_order.payout_request_id

```

since im using sqlite, i will now need to delete the entire db, hopefully i could get away with just a table deletion,

eoliveros commented:
> just pulled the update. everytime i refresh the dashboard on broker order, im getting the following:
> 
> ```
> sqlalchemy.exc.OperationalError: (sqlite3.OperationalError) no such column: broker_order.payout_request_id
> ```
> 
> since im using sqlite, i will now need to delete the entire db, hopefully i could get away with just a table deletion.

deleting the table broker_order, im now getting another error, this time its missing the exchange_deposit.txid
deleting the table exchange_deposit.

eoliveros commented:
cant import the filters from admin into the models.py
```
ImportError: cannot import name 'Role' from partially initialized module 'models' (most likely due to a circular import) (/home/eoliveros/git/beryllium/models.py)
```
so im not sure how i could apply create the filters for the broker_order.

djpnewton commented:
all the filters for the admin pages are in admin.py now
